code/MimeUploadValidator.php

```diff
@@ -88,6 +88,7 @@ public function getExpectedMimeTypes($tmpFile)
         // Get the mime types and their variations from mimevalidator
         $knownMimes = Config::inst()->get(get_class($this), 'MimeTypes');
         if (isset($knownMimes[$extension])) {
+          $knownMimes[$extension] = (array) $knownMimes[$extension];
```

Comment:
It's better style to put the modified list into another variable, rather than modifying the underlying list.